explorer/syntax/parser.ypp

```diff
@@ -326,34 +326,41 @@
 %%
 input: package_directive import_directives declaration_list
     {
-      *ast = AST({.package = $1.first,
-                  .is_api = $1.second,
-                  .imports = std::move($2),
-                  .declarations = std::move($3)});
+      *ast = AST({.package = $[package_directive].first,
+                  .is_api = $[package_directive].second,
+                  .imports = std::move($[import_directives]),
+                  .declarations = std::move($[declaration_list])});
     }
 ;
 package_directive:
   PACKAGE identifier optional_library_path api_or_impl SEMICOLON
-    { $$ = {LibraryName({.package = $2, .path = $3}), $4}; }
+    {
+      $$ = {LibraryName(
+                {.package = $[identifier], .path = $[optional_library_path]}),
+            $[api_or_impl]};
+    }
 ;
 import_directive:
   IMPORT identifier optional_library_path SEMICOLON
-    { $$ = LibraryName({.package = $2, .path = $3}); }
+    {
+      $$ = LibraryName(
+          {.package = $[identifier], .path = $[optional_library_path]});
+    }
 ;
 import_directives:
   // Empty
     { $$ = std::vector<LibraryName>(); }
-| import_directives import_directive
+| import_directives[accumulated] import_directive
```

Comment:
Renamed to `accumulated_list` -- I prefer a standard name for two reasons:

- It provides a consistent naming pattern for a standard usage pattern.
- The use of the name is always nearby (two lines down, separated only by a `{` line), so it's easy to check what the name is coming from.
 
I'm hoping that the additional `_list` clarifies your "not clearly an alias" concern. Some of the cases would be `accumulated_list` even with your suggested "`accumulated_X` where X is the last word of the aliased grammar" naming scheme, so if it still isn't clear enough that this is an alias syntax, it may be a challenging issue to address without adding comments explaining bison grammar.

Reply:
> Renamed to `accumulated_list` -- I prefer a standard name for two reasons:
> 
> * It provides a consistent naming pattern for a standard usage pattern.

Personally I feel like the `accumulated_` prefix provides enough signal that these are all instances of the same pattern, but of course that's subjective.

> * The use of the name is always nearby (two lines down, separated only by a `{` line), so it's easy to check what the name is coming from.

I agree, but that seems like it applies equally to all the options here, and in any event it only applies once you understand the declaration syntax, but my concern here is readers who don't.

> I'm hoping that the additional `_list` clarifies your "not clearly an alias" concern. 

I don't think this addresses my concern -- the repetition of `directives`/`list`/etc between the original name and the alias is central to my suggestion, because it provides an immediate, almost syntactic cue that these two identifiers are talking about the same thing.

I think using `accumulated_list` everywhere might work if we could assume the reader is already aware that this component is acting as an accumulator, but I don't think we can assume that. To recognize that, the reader needs to understand the intended meaning of this rule, or at least notice that the rule is recursive, but the syntactic confusion I'm worried about would happen at a very early stage of reading the code, before the reader is in a position to understand the meaning of the rule. Consequently, any cues we use to address the confusion will need to be very superficial.

> Some of the cases would be `accumulated_list` even with your suggested "`accumulated_X` where X is the last word of the aliased grammar" naming scheme, so if it still isn't clear enough that this is an alias syntax, it may be a challenging issue to address without adding comments explaining bison grammar.

I'm not saying that `accumulated_list` is never clear enough, I'm just saying that it's not clear enough when the original identifier doesn't end with `list`. The repetition is load-bearing.

Reply:
I was thinking about this, and went ahead and switched all aliases so that they're simple prefixes, for consistency. `child_`, `accumulated_`, `lhs_`, and `rhs_` are now the standard prefixes in use to disambiguate.

Reply:
Yeah, this is even better than what I suggested.